frameworks/spark/src/main/java/com/criteo/hadoop/garmadon/spark/listener/GarmadonSparkStorageStatusListener.java

```diff
@@ -0,0 +1,314 @@
+                    distribution.diskUsed = addDelta(distribution.diskUsed, diskDelta);
+                    distribution.memoryUsed = addDelta(distribution.memoryUsed, memoryDelta);
+                    distribution.offHeapMemoryUsed = addDelta(distribution.offHeapMemoryUsed, offHeapMemoryDelta);
+                } else {
+                    //no block on the executor, we can remove the distribution
+                    rddInfo.distributions.remove(executorId);
+                }
+
+                //update executor number of blocks
+                exec.rddBlocks += blockDelta;
+            }
+
+            sendRDDStorageStatusEvent(System.currentTimeMillis(), rddInfo);
+        }
+    }
+
+    private void updateExecutorRDDMemoryStatus(String executorId, long diskDelta, long memoryDelta, long offHeapMemoryDelta) {
+        GarmadonExecutorStorageInfo executorInfo = liveExecutors.get(executorId);
+        if (executorInfo != null) {
+            executorInfo.rddOffHeapMemoryUsed = addDelta(executorInfo.rddOffHeapMemoryUsed, offHeapMemoryDelta);
+            executorInfo.rddMemoryUsed = addDelta(executorInfo.rddMemoryUsed, memoryDelta);
+            executorInfo.rddDiskUsed = addDelta(executorInfo.rddDiskUsed, diskDelta);
+
+            sendExecutorStorageStatusEvent(System.currentTimeMillis(), executorId, executorInfo);
+        }
+    }
+
+    private void updateExecutorStreamMemoryStatus(String executorId, long diskDelta, long memoryDelta, long offHeapMemoryDelta) {
+        GarmadonExecutorStorageInfo executorInfo = liveExecutors.get(executorId);
+        if (executorInfo != null) {
+            executorInfo.streamOffHeapMemoryUsed = addDelta(executorInfo.streamOffHeapMemoryUsed, offHeapMemoryDelta);
+            executorInfo.streamMemoryUsed = addDelta(executorInfo.streamMemoryUsed, memoryDelta);
+            executorInfo.streamDiskUsed = addDelta(executorInfo.streamDiskUsed, diskDelta);
+
+            sendExecutorStorageStatusEvent(System.currentTimeMillis(), executorId, executorInfo);
+        }
+    }
+
+    private void updateExecutorBroadcastMemoryStatus(String executorId, long diskDelta, long memoryDelta, long offHeapMemoryDelta) {
+        GarmadonExecutorStorageInfo executorInfo = liveExecutors.get(executorId);
+        if (executorInfo != null) {
+            executorInfo.broadcastOffHeapMemoryUsed = addDelta(executorInfo.broadcastOffHeapMemoryUsed, offHeapMemoryDelta);
+            executorInfo.broadcastMemoryUsed = addDelta(executorInfo.broadcastMemoryUsed, memoryDelta);
+            executorInfo.broadcastDiskUsed = addDelta(executorInfo.broadcastDiskUsed, diskDelta);
+
+            sendExecutorStorageStatusEvent(System.currentTimeMillis(), executorId, executorInfo);
+        }
+    }
+
+    //Uses same mechanics as spark's AppStatusListener
+    private long addDelta(long value, long delta) {
```

Comment:
It's weird that we need to do this, isn't it? If we end up below 0, wouldn't this mean that we cannot trust the counter?